### PR TITLE
Add a blog post on detecting and dealing with outliers.

### PR DESCRIPTION
**Pull request by brian-brazil**
@juliusv

### CODE REVIEW

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+I have seen more than one attempt by talented engineers to build a system to
+automatically detect and diagnose problems based on time series data. While it
+is certainly possible to get a demonstration working, the data always turned
+out to be too noisy to make this approach work for anything but the simplest of
+real-world systems.
+
+All hope is not lost though. There are some common anomalies which you can
+detect and deal with. The Prometheus [query
+language](../../../../../docs/querying/basics/) gives you the tools to discover
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: At this point an absolute path seems to be the better choice, i.e. `../../../../../docs/querying/basics/` -> `/docs/querying/basics/`.

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+these while avoiding false positives.
+
+## Building a query
+
+A common problem is a small number of servers within a services not
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: -s

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+performing as well as the rest, such as by having increased latency.  While
+the issue may not impact your overall SLA, it is worth having a human look when
+they have a chance.
+
+Let us say that we have `instance:latency_seconds:mean5m` representing the
+average query latency for each instance of a service, calculated from a Summary
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Not sure how well-known the meaning of summary is – maybe link to http://prometheus.io/docs/concepts/metric_types/#summary

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+metric.
+
+A simple way to start would be to look for instances with latency
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: a latency

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+  (
+      avg by (job)(instance:latency_seconds:mean5m) 
+    + on (job)
+      2 * stddev by (job)(instance:latency_seconds:mean5m)
+  )
+```
+
+You try this out, and discover that there are false positives when
+the latencies are very tightly bunched. You add a requirement
+that the instance latency also be 20% above the average:
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: "has to be" might be easier to read.

**Comment by fabxc** [comment]
I like it – grammar check thoughtfully left for @juliusv :)

**Comment by fabxc** [comment]
Maybe sneak in a short mention + link of the query language docs before the example starts.

File: content/blog/2015-06-17-practical-anomaly-detection.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
+---
+title: Practical Anomaly Detection
+created_at: 2015-06-17
+kind: article
+author_name: Brian Brazil
+---
+
+In his [Open Letter To Monitoring/Metrics/Alerting Companies](http://www.kitchensoap.com/2015/05/01/openlettertomonitoringproducts/),
```
Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Italic for the title/link?

Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Done

File: content/blog/2015-06-17-practical-anomaly-detection.md (hunk 5 above)
Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Done

File: content/blog/2015-06-17-practical-anomaly-detection.md (hunk 4 above)
Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Done

File: content/blog/2015-06-17-practical-anomaly-detection.md (hunk 3 above)
Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Done

File: content/blog/2015-06-17-practical-anomaly-detection.md (hunk 2 above)
Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: Done

File: content/blog/2015-06-17-practical-anomaly-detection.md (hunk 1 above)
Comment (brian-brazil) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: We're using relative paths everywhere else - @juliusv is there any reason not to use absolute paths?

Comment (fabxc) [inline content/blog/2015-06-17-practical-anomaly-detection.md]: As we don't have a full-fledged CMS, changing of directory structure is always a point. Relative paths thus make sense within a component, i.e. docs or blog posts. Across them there's little benefit.

**Comment by beorn7** [comment]
Yeah, very nice...

**Comment by juliusv** [comment]
Suggestion: for the minor grammar/wording stuff, would it be ok for me to just push a final commit on top of this at the end? I'll comment a bit on other stuff now.

**Comment by juliusv** [comment]
(for the above to work, it would be nice to push the `anomaly` branch into the official `prometheus/docs` docs repo instead of the fork)

**Comment by brian-brazil** [comment]
See #121

**Pull request closed without merging**